simpleocr/files.py

```diff
@@ -10,80 +9,165 @@
     """
+    def __init__(self, array, debug=False):
+        """
+        :param array: array with image data, must be OpenCV compatible
+        :param debug: if True, debug prints are enabled
+        """
+        self._image = array
+        self._ground = None
+        self._debug = debug
 
-    def __init__(self, image_path):
-        if not os.path.exists(image_path):
-            raise IOError("Image file not found: " + image_path)
-        self.image_path = image_path
-        self.image = cv2.imread(self.image_path)
-        basepath = os.path.splitext(image_path)[0]
-        self.ground_path = try_extensions(GROUND_EXTENSIONS, basepath)
-        if self.ground_path:
-            self.ground = GroundFile(self.ground_path)
-            self.ground.read()
-        else:
-            self.ground_path = basepath + GROUND_EXTENSIONS_DEFAULT
-            self.ground = None
+    def set_ground(self, segments, classes):
+        """ Creates the ground data in memory """
+        if self.is_grounded and self._debug:
+            print("Warning: grounding already grounded Image")
+        self._ground = Ground(segments=segments, classes=classes)
+
+    def remove_ground(self):
+        """ Removes the grounding data in memory for the Image """
+        if not self.is_grounded:
+            print("Warning: removing ground for Image without ground data")
+        self._ground = None
+
+    # These properties prevent the user from altering the attributes stored within
+    # the object and thus emphasize the immutability of the object
+    @property
+    def image(self):
+        return self._image
 
     @property
     def is_grounded(self):
-        """checks if this file is grounded"""
-        return not (self.ground is None)
+        return not (self._ground is None)
+
+    @property
+    def ground(self):
+        return self._ground
+
+
+class ImageFile(Image):
+    """
+    Complete class that contains functions for creation from file,
+    as well as from PIL Images. Also supports grounding in memory.
+    """
+    def __init__(self, path, debug=False):
+        """
+        :param path: path to the image to read, must be valid and absolute
+        :param debug: if True, debug prints are enabled
+        """
+        array = cv2.imread(path)
+        Image.__init__(self, array, debug=debug)
+        self._path = path
+        basepath = os.path.splitext(path)[0]
+        self._ground_path = try_extensions(GROUND_EXTENSIONS, basepath)
+        if self._ground_path:
+            self._ground = GroundFile(self._ground_path)
+            self._ground.read()
+        else:
+            self._ground_path = basepath + GROUND_EXTENSIONS_DEFAULT
+            self._ground = None
 
     def set_ground(self, segments, classes, write_file=False):
-        """creates the ground, saves it to a file"""
-        if self.is_grounded:
+        """ creates the ground, saves it to a file """
+        if self.is_grounded and self._debug:
```

Comment:
But the `ImageFile.set_ground` uses a `GroundFile` instance instead of a `Ground` instance, so the `Image.set_ground` doesn't suffice for the `ImageFile`. The only part that is duplicated is the debug part.